sleepylyze/nrem.py

```diff
@@ -74,7 +74,8 @@ def __init__(self, fname=None, fpath=None, match=None, in_num=None, epoched=Fals
 
             #remove channels
             exclude = ['EOG_L','EOG_R', 'EKG']
-            self.metadata['channels'] = [x[0] for x in self.data.columns if x[0] not in exclude]
+            # create channel attribute
+            self.channels = [x[0] for x in self.data.columns if x[0] not in exclude]
```

Comment:
Instead of combining the "exclude" variable into the "channels" attribute keep them explicit so that the data sources are as clean as possible. The following would work:
```
# create exclude & channels attributes
self.exclude = ['EOG_L', 'EOG_R', 'EKG']
self.channels = [x[0] for x in self.data.columns]
```

see comment in commit 9820c3d2c0b0b2abacfb1c83b409a94a1b7d9b65 for application in the individual functions

Reply:
Kept exclude and channels separate in commit 1438e7f